app/models/concerns/variant_stock.rb

```diff
@@ -20,61 +20,30 @@ module VariantStock
     after_update :save_stock
   end
 
-  # Returns the number of items of the variant available in the stock. When
-  # allowing on demand, it returns infinite.
-  #
-  # Spree computes it as the sum of the count_on_hand of all its stock_items.
+  # Returns the number of items of the variant available.
+  # Spree computes total_on_hand as the sum of the count_on_hand of all its stock_items.
   #
   # @return [Float|Integer]
   def on_hand
     warn_deprecation(__method__, '#total_on_hand')
 
-    if on_demand
-      Float::INFINITY
```

Comment:
This is a step backwards. We changed this because Spree introduced this logic of returning infinity. This change would make us less compatible with Spree again. I would prefer to adjust the cart logic or the serializer to deal with infinity. The `9999` value in the cart logic is just a workaround for not dealing with infinity properly.

Reply:
I think this is a step forward Maikel. I will explain.
Returning Infinity is poor, it's magic, the client should get specific on_hand and on_demand values all the time and the client should decide what to do, no magic.

Also, one important detail for this logic is that in spree, and soon in ofn, on_demand means backorderable: that means that even if the item is backorderable (which will be the on_demand field), the on_hand value will still be used. Basically stock can go to negative.

So, this is a move in the direction of making on_demand equal to backorderable and also it removes magic.

did I convince you? :-D

Reply:
"The `9999` value in the cart logic is just a workaround"
I am not too worried with client side code here, we can fix it later, here we are changing OFN core behaviour, and I think that it is important to be very clear and simple, i.e., not mixing on_hand and on_demand in the core. Leaving that up to the user.

Reply:
I really liked the infinity solution as it seemed mathematically correct and solving all the problems. But you convinced me with the different use of on_hand. There are two different uses that Spree started to separate:

- How many can I order? Infinity is right here.
- How many are in stock? That's a finite number of the stock item.

And on_hand sounds a bit more like the answer to the second question.